phoenix-core/src/it/java/org/apache/phoenix/end2end/JsonEachFunctionIT.java

```
@@ -0,0 +1,107 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.end2end;
+
+import org.apache.phoenix.schema.types.PhoenixArray;
+import org.apache.phoenix.util.PropertiesUtil;
+import org.junit.Test;
+
+import java.sql.*;
+import java.util.Properties;
+
+import static org.apache.phoenix.util.TestUtil.TEST_PROPERTIES;
+import static org.junit.Assert.*;
+
+/**
+ * End to end test for {@link org.apache.phoenix.expression.function.JsonEachFunction}.
+ *
+ */
+public class JsonEachFunctionIT extends BaseHBaseManagedTimeIT {
+
+    @Test
+      public void testJsonEach() throws Exception {
+        Connection conn = getConnection();
+
+        try{
+            String json =
+                    "{\"f2\":{\"f3\":\"value\"},\"f4\":{\"f5\":99,\"f6\":[1,true,\"foo\"]},\"f7\":true}";
+            populateJsonTable(conn, json, "valueOne");
+
+            String selectQuery = "SELECT json_each(col1) FROM testJson WHERE pk = 'valueOne'";
+            PreparedStatement stmt = conn.prepareStatement(selectQuery);
+            ResultSet rs = stmt.executeQuery();
+            assertTrue(rs.next());
+            String[] strArr = new String[]{
+                    "f2,{\"f3\":\"value\"}",
+                    "f4,{\"f5\":99,\"f6\":[1,true,\"foo\"]}",
+                    "f7,true"};
+            Array array = conn.createArrayOf("VARCHAR", strArr);
+            PhoenixArray resultArray = (PhoenixArray) rs.getArray(1);
+            assertEquals("JsonEach return data is not as expected.", resultArray,
+                    array);
+            assertFalse(rs.next());
+
+        } finally {
+            conn.close();
+        }
+    }
+
+    @Test
+    public void testJsonEachWithNullKey() throws Exception {
+        Connection conn = getConnection();
+
+        try{
+            String json =
+                    "100";
```

This should throw an exception because 100 is not valid json.